### PR TITLE
FIX: Hook to user_seen event to auto-join users on their first visit.

### PR DESCRIPTION
**Pull request by romanrizzi**

### CODE REVIEW

File: spec/plugin_spec.rb [hunk 1]
```diff
@@ -254,35 +254,25 @@ def assert_user_following_state(user, channel, following:)
 
     describe "when a user is created" do
       fab!(:category) { Fabricate(:category) }
-      fab!(:user) { Fabricate(:user) }
+      let(:user) { Fabricate(:user, last_seen_at: nil, first_seen_at: nil) }
```
Comment (jjaffeux) [inline spec/plugin_spec.rb:257]: do we need let now?

Comment (romanrizzi) [reply spec/plugin_spec.rb:257]: Yes. Re-using the user with `fab!` makes the first test flaky.

File: spec/plugin_spec.rb [hunk 2]
```diff
@@ -254,35 +254,25 @@ def assert_user_following_state(user, channel, following:)
...
         assert_user_following_state(user, channel, following: false)
       end
 
       it "does nothing if auto-join is disabled" do
         channel.update!(auto_join_users: false)
 
-        user.update!(last_seen_at: 1.minute.ago)
+        user.update_last_seen!
 
         assert_user_following_state(user, channel, following: false)
```
Comment (jjaffeux) [inline spec/plugin_spec.rb:277]: Maybe we need a test with `user.update!(first_seen_at: Time.now)`? Or do we have it somewhere already?

Comment (romanrizzi) [reply spec/plugin_spec.rb:277]: I believe the spec in line 265 covers that scenario

**Review by jjaffeux — APPROVED** [review]

**Pull request merged**